source/MaterialXTest/MaterialXRuntime/Runtime.cpp

```diff
@@ -1075,6 +1076,42 @@ TEST_CASE("Runtime: DefaultLook", "[runtime]")
     fileIo.read("emptyLook.mtlx", lookSearchPath);
 }
 
+TEST_CASE("Runtime: namespaced definitions", "[runtime]")
+{
+    mx::RtScopedApiHandle api;
+
+    mx::FilePath librariesPath = mx::FilePath::getCurrentPath() / mx::FilePath("libraries");
+    mx::FilePathVec rootPaths;
+    rootPaths.push_back(librariesPath);
+    mx::FileSearchPath searchPath(librariesPath);
+    mx::FilePathVec childFolders;
+    mx::getSubdirectories(rootPaths, searchPath, childFolders);
+    for (const auto& childFolder : childFolders)
+    {
+        searchPath.append(childFolder);
+    }
+    api->setSearchPath(searchPath);
+    api->loadLibrary(STDLIB);
+    api->loadLibrary(PBRLIB);
+    api->loadLibrary(BXDFLIB);
+    api->loadLibrary(ADSKLIB);
+
+    mx::RtStagePtr defaultStage = api->createStage(mx::RtToken("defaultStage"));
+    defaultStage->addReference(api->getLibrary());
+
+    mx::FileSearchPath adskTestPath(mx::FilePath::getCurrentPath() /
+        "resources" /
+        "Materials" / 
+        "TestSuite" / 
+        "adsklib");
+
+    mx::RtFileIo fileIo(defaultStage);
+
+    mx::RtReadOptions options;
+    options.validateDocument = false;
```

Comment:
Need to set this to false as the Document does not have all the definitions, only masterPrim. 
The RtFile readDocument() method will perform the masterprim nodedef check (w/ above fixes).